dftimewolf/lib/processors/grepper.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+"""Processes artifacts using local grep ."""
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import os
+import tempfile
+import re
+import mimetypes
+import PyPDF2
+
+from dftimewolf.lib.module import BaseModule
+
+
+class GrepperSearch(BaseModule):
+  """Processes a list of file paths with to search for
+  specific keywords.
+
+  input: A file path to process, and a list of keywords to search for
+  output: filepath and keyword match, to stdout (final_output).
+  """
+
+  def __init__(self, state):
+    super(GrepperSearch, self).__init__(state)
+    self._keywords = None
+    self._output_path = None
+    self._final_output = None
+
+  def setup(self, keywords=None):  # pylint: disable=arguments-differ
+    """Sets up the _keywords attribute.
+
+    Args:
+      keywords: pipe separated list of keyword to search
+    """
+    self._keywords = keywords
+    self._output_path = tempfile.mkdtemp()
+
+  def cleanup(self):
+    pass
+
+  def process(self):
+    """Execute the grep command"""
+
+    for _, path in self.state.input:
+      log_file_path = os.path.join(self._output_path, 'grepper.log')
+      print('Log file: {0:s}'.format(log_file_path))
+
+      print('Walking through dir (absolute) = ' + os.path.abspath(path))
+      try:
+        for root, _, files in os.walk(path):
+          for filename in files:
+            found = set()
+            fullpath = os.path.abspath(root) + '/' + filename
+            if mimetypes.guess_type(filename)[0] == 'application/pdf':
+              found = self.grepPDF(fullpath)
+            else:
+              with open(fullpath, 'r') as fp:
+                for line in fp:
+                  found.update(set(x.lower() for x in re.findall(
+                      self._keywords, line, re.IGNORECASE)))
+            if filter(None, found):
+              output = path + '/' + filename + ':' + ','.join(
```

Comment:
Please use a format string here (e.g. `'{0:s}/{1:s}:{2:s}'.format(path, filename, ','.join(filter(None, found))`)

Also, TIL about `filter(None, <list>)`! I would've done `[item for item in <list> if item]`

Reply:
fixed, both the string the the suggestion. thanks